SourceX/DiabloUI/mainmenu.cpp

```diff
@@ -47,13 +47,20 @@ void mainmenu_Free()
 
 BOOL UiMainMenuDialog(char *name, int *pdwResult, void(*fnSound)(char *file), int a4)
 {
+	BOOL played = false;
 	mainmenu_Load(name, fnSound);
 
 	while (MainMenuResult == 0) {
+		if (dwRepIntro > 0 && GetTickCount() >= dwRepIntro) {
+			played = true;
```

Comment:
This could probably be done picking the "Replay Intro" option. Once the video is done play the game will redirect to the manu, so that should simplify a few things.
```suggestion
			MainMenuResult = 2;
```

Reply:
Removed both the `dwRepIntro > 0` and `BlackPalette` skip now that it's guaranteed to restart the counter when entering the menu.